osu.Game/Overlays/Profile/Sections/Historical/ProfileLineChart.cs

```diff
@@ -0,0 +1,242 @@
+                                        {
+                                            RelativeSizeAxes = Axes.Both
+                                        },
+                                        columnLinesContainer = new Container<TickLine>
+                                        {
+                                            RelativeSizeAxes = Axes.Both
+                                        }
+                                    }
+                                },
+                                graph = new UserHistoryGraph
+                                {
+                                    RelativeSizeAxes = Axes.Both
+                                }
+                            }
+                        }
+                    },
+                    new[]
+                    {
+                        Empty(),
+                        columnTicksContainer = new Container<TickText>
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            AutoSizeAxes = Axes.Y,
+                            Padding = new MarginPadding { Top = 10 }
+                        }
+                    }
+                }
+            };
+        }
+
+        private void createRowTicks()
+        {
+            rowTicksContainer.Clear();
+            rowLinesContainer.Clear();
+
+            var min = values.Select(v => v.Count).Min();
+            var max = values.Select(v => v.Count).Max();
+
+            var niceRange = niceNumber(max - min, false);
+            var niceTick = niceNumber(niceRange / (6 - 1), true);
+
+            double rollingRow = min;
+
+            while (rollingRow <= max)
+            {
+                var y = -Interpolation.ValueAt(rollingRow, 0, 1f, min, max);
+
+                rowTicksContainer.Add(new TickText
+                {
+                    Anchor = Anchor.BottomRight,
+                    Origin = Anchor.CentreRight,
+                    RelativePositionAxes = Axes.Y,
+                    Margin = new MarginPadding { Right = 3 },
+                    Text = rollingRow.ToString("N0"),
+                    Font = OsuFont.GetFont(size: 12),
+                    Y = y
+                });
+
+                rowLinesContainer.Add(new TickLine
+                {
+                    Anchor = Anchor.BottomRight,
+                    Origin = Anchor.CentreRight,
+                    RelativeSizeAxes = Axes.X,
+                    RelativePositionAxes = Axes.Y,
+                    Height = 1,
+                    Y = y
+                });
+
+                rollingRow += niceTick;
+            }
+        }
+
+        private void createColumnTicks()
+        {
+            columnTicksContainer.Clear();
+            columnLinesContainer.Clear();
+
+            var min = values.Select(v => v.Date).Min().ToOADate();
+            var max = values.Select(v => v.Date).Max().ToOADate();
```

Comment:
empty sequences are not checked here and will crash.

also calculating the difference would be done more sanely not by taking this eldritch `ToOADate` thing (why is that even a thing?) but via `TimeSpan.TotalDays` or similar.